### ISSUE TITLE
recursive diff

### ISSUE THREAD

**Issue by davidkraus** [Kuniwak/MirrorDiffKit#10]
Hey,

we started using your library and liked the experience so far, but now we have the following problem.

We have data types like this:
```
struct Event {
  let title:String
  let data:[String]
}
```
In our tests the data array consists of more complex types and can get very large. Mostly they are the same but on one position the data is different. What we now get is a huge output telling us that data is different but not exactly where this is. 

```
struct Event {
  title: \"I\'m not changed\""
  - data: [\"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"I\'m deleted\"]"
  + data: [\"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"nothing\", \"I\'m inserted\"]"
}
```

So i tried to change that by calling Diffable.diff inside of DiffableDictionary (line 76 to 78) like so `result[key] = Diffable.diff(between: lv, and: rv)` to go deeper into the data structures.

this changes the output to:

```
struct Event {
  title: \"I\'m not changed\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  data: \"nothing\"
  - data: \"I\'m deleted\"
  + data: \"I\'m inserted\"
}
```

which is way clearer but looks a little strange

Do you have any plans to support a deeper diff or know how we could tackle this problem?

**Comment by Kuniwak** [comment]
I think the following format style is better.

```swift
struct Event {
  title: "I'm not changed"
  data: [
    "nothing",
    "nothing",
    "nothing",
    "nothing",
    "nothing",
    "nothing",
    "nothing",
    "nothing",
    "nothing",
    "nothing",
  - "I'm deleted"
  + "I'm inserted"
  ]
}
```

So, I'm gonna try to implement it.

**Comment by Kuniwak** [comment]
I shipped this feature at v1.0.2.
Please try it.

**Comment by davidkraus** [comment]
Hey thanks, that was quick

i did a short test and it looks much better now 👍
